alicloud/resource_alicloud_cr_ee_instance.go

```diff
@@ -0,0 +1,185 @@
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceAlicloudCrEEInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudCrEEInstanceCreate,
+		Read:   resourceAlicloudCrEEInstanceRead,
+		Delete: resourceAlicloudCrEEInstanceDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"payment_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Subscription"}, false),
+				Default:      "Subscription",
+			},
+			"period": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ForceNew: true,
+			},
+			"renew_period": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+			},
+			"renewal_status": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"AutoRenewal", "ManualRenewal"}, false),
+			},
+			"instance_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"Basic", "Standard", "Advanced"}, false),
+			},
+			"instance_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"custom_oss_bucket": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_time": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"instance_specification": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAlicloudCrEEInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	request := bssopenapi.CreateCreateInstanceRequest()
+	request.Scheme = "https"
+	request.RegionId = "cn-hangzhou"
+	request.ProductCode = "acr"
+	request.ProductType = "acr_ee_public_cn"
+	request.SubscriptionType = d.Get("payment_type").(string)
+	request.Period = requests.NewInteger(d.Get("period").(int))
+	if v, ok := d.GetOk("renew_period"); ok {
+		request.RenewPeriod = requests.NewInteger(v.(int))
+	}
+	if v, ok := d.GetOk("renewal_status"); ok {
+		request.RenewalStatus = v.(string)
+	}
+
+	parameters := []bssopenapi.CreateInstanceParameter{
+		{
+			Code:  "InstanceType",
+			Value: d.Get("instance_type").(string),
+		},
+		{
+			Code:  "InstanceName",
+			Value: d.Get("instance_name").(string),
+		},
+		{
+			Code:  "Region",
+			Value: client.RegionId,
+		},
+	}
+	if v, ok := d.GetOk("custom_oss_bucket"); ok {
+		parameters = append(parameters, bssopenapi.CreateInstanceParameter{
+			Code:  "DefaultOssBucket",
+			Value: "false",
```

Comment:
The `custom_oss_bucket ` type is string in the Schema, not bool. Please keep it sync.